examples/angularjs/js/services/gifFetch.js

```diff
@@ -0,0 +1,42 @@
+/*global angular */
+
+/**
+ * Services that retrieve gifs from the API's on the basis of
+ * the todo's title.
+ *
+ * They both follow the same API, returning promises for all changes to the
+ * model.
+ */
+angular.module('todomvc')
+	.factory('gifFetch', function ($resource) {
+		'use strict';
+
+		var gif = {
+
+            defURL: 'https://marshallwritingcenter.files.wordpress.com/2015/04/gif2.gif',
+
+			api: $resource('https://api.tenor.com/v1/search', {
+                limit: 1,
+                tag: '',
+            }),
+
+			get: function (tag) {
+                if (_.size(tag) === 0) {
+                    return gif.defURL;
+                }
+                // "eggs" give better result than "Buy eggs tomorrow"
+                const parsed = nlp(tag);
```

Comment:
Great idea to refine text to nouns.

This should go into a separate service. This is again separation of concern. Having two services with clear responsibilities is non ambiguous: `GifsService` and `NLPService`.